mongodb/vibe/db/mongo/collection.d

```diff
@@ -253,19 +254,59 @@ struct MongoCollection {
 	}
 
 	/**
-		Replaces at most single document within the collection based on the filter.
+		Replaces at most single document within the collection based on the
+		filter.
+
+		It's recommended to use the ReplaceOptions overload, but UpdateOptions
+		can be used as well. Note that the extra options inside UpdateOptions
+		may have no effect, possible warnings for this may only be handled by
+		MongoDB.
 
 		See_Also: $(LINK https://www.mongodb.com/docs/manual/reference/method/db.collection.replaceOne/#mongodb-method-db.collection.replaceOne)
 
 		Standards: $(LINK https://www.mongodb.com/docs/manual/reference/command/update/)
 	*/
+	UpdateResult replaceOne(T, U)(T filter, U replacement, ReplaceOptions options)
+	@safe {
+		UpdateOptions uoptions;
+		static foreach (f; FieldNameTuple!ReplaceOptions)
+			__traits(getMember, uoptions, f) = __traits(getMember, options, f);
+		Bson opts = Bson.emptyObject;
+		opts["multi"] = Bson(false);
+		return updateImpl([filter], [replacement], [opts], uoptions, true, false);
+	}
+
+	/// ditto
 	UpdateResult replaceOne(T, U)(T filter, U replacement, UpdateOptions options = UpdateOptions.init)
 	@safe {
 		Bson opts = Bson.emptyObject;
 		opts["multi"] = Bson(false);
 		return updateImpl([filter], [replacement], [opts], options, true, false);
 	}
 
+	///
+	@safe deprecated unittest {
```

Comment:
```suggestion
	@safe unittest {
```

Looks like this is just a copy-paste leftover?

Reply:
yes